### PR TITLE
[#8572] Run twistedchecker with Travis-CI

### PR DESCRIPTION
**Pull request by adiroiban**
https://twistedmatrix.com/trac/ticket/8572

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 language: python
 sudo: false
 
+notifications:
```
Comment (glyph) [inline .travis.yml]: This seems irrelevant to the change at hand here?

File: bin/admin/twistedchecker-trunk-diff.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env sh
+#
+# Helper for running twistedchecker and reporting only errors that are part
```
Comment (glyph) [inline bin/admin/twistedchecker-trunk-diff.sh]: If we are going to have helpers that developers are expected to run, they should be in Python, not shell scripts.  Shell should only be completely travis-specific stuff.  Shell scripts, among other problems, don't run on Windows.

File: tox.ini [hunk 3]
```diff
@@ -73,7 +77,7 @@ changedir={toxinidir}
 [tox:travis]
 ; This section is used by tox-travis
 ; https://pypi.python.org/pypi/tox-travis
-2.7 = py27-coverage-posix,py27-nomodules-posix,topfile,narrativedocs,pyflakes
+2.7 = py27-coverage-posix,py27-nomodules-posix,topfile,narrativedocs,pyflakes,txchecker-travis
```
Comment (glyph) [inline tox.ini]: We should really get rid of tox-travis and have a proper build matrix; but that is for a separate ticket, I think.

File: bin/admin/twistedchecker-trunk-diff.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+# Call it as:
+# * SCRIPT_NAME twisted
+# * SCRIPT_NAME twisted/words/
+# * SCRIPT_NAME twisted.words
+
+target=$1
+
+# Since for unknown modules twistedchecker will return the same error, the
+# diff will fail to detect that we are trying to check an invalid path or
+# module.
```
Comment (glyph) [inline bin/admin/twistedchecker-trunk-diff.sh]: This sounds like a twistedchecker bug, something that should be fixed upstream.

File: tox.ini (hunk 3 above)
Comment (adiroiban) [inline tox.ini]: It is https://twistedmatrix.com/trac/ticket/8535

File: bin/admin/twistedchecker-trunk-diff.sh (hunk 4 above)
Comment (adiroiban) [inline bin/admin/twistedchecker-trunk-diff.sh]: I have created an upstream ticket

File: bin/admin/twistedchecker-trunk-diff.sh (hunk 2 above)
Comment (adiroiban) [inline bin/admin/twistedchecker-trunk-diff.sh]: I have moved it in .travis, but I think that we are just fooling ourselves , as I will continue to use it from .travis directory as it better than nothing and better than doing a commit and push to see if my changes are introducing any errors :)

File: .travis.yml (hunk 1 above)
Comment (adiroiban) [inline .travis.yml]: ok. reverted.
